lib/spack/spack/test/cmd/spec.py

```diff
@@ -50,6 +52,25 @@ def test_spec_json():
     assert 'mpich' in mpileaks
 
 
+def test_spec_format(database, config):
+    output = spec('--format', '{name}-{^mpi.name}', 'mpileaks')
+    assert output.rstrip('\n') == "mpileaks-mpich"
+
+    output = spec('--format', '{name}-{version}-{compiler.name}-{^mpi.name}',
+                  'mpileaks')
```

Comment:
My concern with any checks of versions, compilers and providers is that they are tied to the current package and test configuration.  That is, if the package changes, the preferred/latest version will change; if the default compiler or provider changes, then that will affect those entries.

You could address this by changing `'mpileaks'` to `'mpileaks@2.3'` and adding `, '%gcc', '^mpich'` to make sure the remaining values